repos/system_upgrade/el7toel8/actors/ziplcheckmultiplerescueentries/tests/test_ziplcheckmultiplerescueentries.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+
+from leapp import reporting
+from leapp.libraries.actor import ziplcheckmultiplerescueentries
+from leapp.libraries.actor.ziplcheckmultiplerescueentries import inhibit_if_multiple_rescue_entries_present
+from leapp.libraries.common.testutils import create_report_mocked, CurrentActorMocked
+from leapp.libraries.common.config import architecture
+from leapp.libraries.stdlib import api
+from leapp.models import BootEntry
+from leapp.snactor.fixture import current_actor_context
+
+
+def test_inhibition_multiple_rescue_entries_present(monkeypatch):
+    """Tests whether the upgrade process is inhibited when multiple rescue boot entries are present."""
+    mocked_report = create_report_mocked()
+    monkeypatch.setattr(reporting, 'create_report', mocked_report)
+
+    boot_entries = [
+        BootEntry(title='entry_1'),
+        BootEntry(title='entry_1_Rescue'),
+        BootEntry(title='entry_2'),
+        BootEntry(title='entry_2_rescue-ver2.3'),  # Typically the `rescue` substring surrounded
+    ]
+
+    inhibit_if_multiple_rescue_entries_present(boot_entries)
+
+    assert mocked_report.called, 'Report should be created when multiple rescue entries are present.'
+
+    fail_description = 'The correct rescue entries are not present in the report summary.'
+    report_summary = mocked_report.report_fields['summary']
+    for expected_rescue_entry in ['entry_1_Rescue', 'entry_2_rescue-ver2.3']:
+        assert expected_rescue_entry in report_summary, fail_description
+
+    fail_description = 'Upgrade should be inhibited on multiple rescue entries.'
+    assert 'inhibitor' in mocked_report.report_fields['flags'], fail_description
+
+
+def test_inhibition_multiple_rescue_entries_not_present(monkeypatch):
+    """Tests whether the upgrade process is not inhibited when multiple rescue boot entries are not present."""
+    monkeypatch.setattr(reporting, 'create_report', create_report_mocked())
+
+    boot_entries = [
+        BootEntry(title='entry_1'),
+        BootEntry(title='entry_2'),
+        BootEntry(title='entry_2_rescue-ver2.3'),  # Typically the `rescue` substring surrounded
+    ]
+
+    inhibit_if_multiple_rescue_entries_present(boot_entries)
+
+    assert not reporting.create_report.called, 'Report was created, even if multiple rescue entries were not present.'
+
+
+def test_actor_does_nothing_on_non_s390x_architecture(monkeypatch, current_actor_context):
+    """Tests whether the actor doesn't perform different architectures than s390x."""
+    def matches_architecture_mocked(arch):
+        if arch == architecture.ARCH_X86_64:
+            return True
+        return False
```

Comment:
The test will pass only on intel arch. In case someone execute it on a non-intel arch, test is broken. In this case, the correct way of testing would be telling what system the actor is expected to be executed during the test. E.g. tell the actor the system is s390x and see the result. Of course for such a test you cannot use the component test (`current_actor_context`), but you have to write the unit test instead.